pkg/cmd/pr/checkout/checkout.go

```diff
@@ -71,15 +87,47 @@ func NewCmdCheckout(f *cmdutil.Factory, runF func(*CheckoutOptions) error) *cobr
 }
 
 func checkoutRun(opts *CheckoutOptions) error {
-	findOptions := shared.FindOptions{
-		Selector: opts.SelectorArg,
-		Fields:   []string{"number", "headRefName", "headRepository", "headRepositoryOwner", "isCrossRepository", "maintainerCanModify"},
-	}
-	pr, baseRepo, err := opts.Finder.Find(findOptions)
-	if err != nil {
-		return err
-	}
+	var (
+		baseRepo ghrepo.Interface
+		pr       *api.PullRequest
+		err      error
+	)
+
+	switch {
+	case opts.SelectorArg != "":
+		pr, baseRepo, err = opts.Finder.Find(shared.FindOptions{
+			Selector: opts.SelectorArg,
+			Fields: []string{
+				"number",
+				"headRefName",
+				"headRepository",
+				"headRepositoryOwner",
+				"isCrossRepository",
+				"maintainerCanModify",
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+	default:
+		if !opts.IO.CanPrompt() {
+			return cmdutil.FlagErrorf("must provide a pull request number (or URL or branch) when not running interactively")
+		}
```

Comment:
I think we should move this logic into the `RunE` section of the Cobra setup to align with some of the other commands. We could also reword the error message to align with the other commands:

```
pull request number, URL, or branch required when not running interactively
```

Here's some examples:

https://github.com/cli/cli/blob/9c550992ea4f207fa293cf0de00b633b47cff4c9/pkg/cmd/run/view/view.go#L155-L163

https://github.com/cli/cli/blob/9c550992ea4f207fa293cf0de00b633b47cff4c9/pkg/cmd/workflow/run/run.go#L96-L102

Reply:
Note: this is also resolved through my suggested patch in https://github.com/cli/cli/pull/9868#issuecomment-2513244933